src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs

```diff
@@ -369,6 +369,17 @@ private async Task<ImmutableArray<Diagnostic>> GetAnalyzerDiagnosticsWithoutStat
             return _analysisResultBuilder.GetDiagnostics(analysisScope, getLocalDiagnostics: true, getNonLocalDiagnostics: true);
         }
 
+        private static AnalyzerDriver CreateDriverForComputingDiagnosticsWithoutStateTracking(Compilation compilation, ImmutableArray<DiagnosticAnalyzer> analyzers)
+        {
+            // Create and attach a new driver instance to compilation, do not used a pooled instance from '_driverPool'.
+            // Additionally, use a new AnalyzerManager and don't use the analyzer manager instance
+            // stored onto the field '_analyzerManager'.
+            // Pooled _driverPool and shared _analyzerManager is used for partial/state-based analysis, and
+            // we want to compute diagnostics without any state tracking here.
+
+            return compilation.CreateAnalyzerDriver(analyzers, new AnalyzerManager(analyzers), severityFilter: SeverityFilter.None);
```

Comment:
@dotnet/roslyn-compiler This is the only compiler change in this PR. We need to ensure that we do not share the same instance of [AnalyzerManager](https://github.com/dotnet/roslyn/blob/master/src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerManager.cs) across partial/tree based analysis and complete/full compilation analysis. Before this PR, IDE executed open file analysis within devenv.exe and project level analysis in OOP process, so we never shared same `CompilationWithAnalyzers` instance for diagnostic computation. After this change, we now share same `CompilationWithAnalyzers` instance for all diagnostic computation (light bulb, active/open files and entire project) in OOP. Sharing the same analyzer manager instance was leading to corrupted state within that instance, which is now avoided.